ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/DynamoDbTest.kt

```diff
@@ -0,0 +1,219 @@
+import java.nio.file.Path
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.UUID
+import kotlin.random.Random
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class DynamoDbTest {
+
+    private val date = LocalDate.now().format(DateTimeFormatter.BASIC_ISO_DATE)
+    private val tableName = "uitest-$date-${UUID.randomUUID()}"
+    private val dynamo = "DynamoDB"
+    private val otherIndex = "OtherIndex"
+
+    private lateinit var client: DynamoDbClient
+    private lateinit var items: List<AllTypesType>
+
+    @TempDir
+    lateinit var tempDir: Path
+
+    @BeforeAll
+    fun setUp() {
+        log.info("Creating table $tableName")
+        client = DynamoDbClient.builder().region(Region.US_WEST_2).build()
+        createAndPopulateTable()
+        log.info("Table created")
+    }
+
+    @AfterAll
+    fun cleanup() {
+        log.info("Running final cleanup")
+        client.deleteTable {
+            it.tableName(tableName)
+        }
+        client.waiter().waitUntilTableNotExists { it.tableName(tableName) }
+        log.info("Deleted table $tableName")
+    }
+
+    @Test
+    @CoreTest
+    fun testDynamoDbTableViewer() = uiTest {
+        welcomeFrame {
+            openFolder(tempDir)
+        }
+
+        idea {
+            waitForBackgroundTasks()
+
+            step("Open table") {
+                awsExplorer {
+                    expandExplorerNode(dynamo)
+                    doubleClickExplorer(dynamo, tableName)
+                }
+
+                step("Check all data loaded") {
+                    step("Wait for data to load") {
+                        waitFor { findByXpath("//div[@class='JBTableHeader']").retrieveData().textDataList.map { it.text }.isNotEmpty() }
+                    }
+                    step("Check full table scan (default index)") {
+                        val columnCount = step("Check columns") {
+                            val cols = findByXpath("//div[@class='JBTableHeader']").retrieveData().textDataList.map { it.text }
+                            // Can't do a straight-up string match because the column headers may be truncated for space
+                            assertThat(cols).satisfiesExactlyInAnyOrder(
+                                { assertThat(it).startsWith("Num") },
+                                { assertThat(it).startsWith("Str") },
+                                { assertThat(it).startsWith("Ter") },
+                                { assertThat(it).startsWith("Boo") },
+                                { assertThat(it).startsWith("Int") },
+                                { assertThat(it).startsWith("Str") },
+                                { assertThat(it).startsWith("Num") },
+                                { assertThat(it).startsWith("Map") },
+                                { assertThat(it).startsWith("Str") },
+                                { assertThat(it).startsWith("Num") }
+                            )
```

Comment:
Fun....

Ya the correct way would be replacing the text renderers with a custom text extractor I believe, and that code has to be upstreamed to be done easily.

This will work for now

Reply:
Not sure its worth doing that for the sake of UI tests only

Reply:
Ya, I have upstreamed a few fixes and features. If we make a full JTableFixture one day, that is when I would upstream it